src/vodozemac/pk_encryption.rs

```diff
@@ -0,0 +1,183 @@
+//! This module provides public-key encryption and decryption functionality
+//! using the `vodozemac` cryptographic library.
+//!
+//! This implements the m.megolm_backup.v1.curve25519-aes-sha2 algorithm
+//! described in the Matrix [spec].
+//!
+//! More details can be found in the official [vodozemac documentation].
+//!
+//! [spec]: https://spec.matrix.org/v1.11/client-server-api/#backup-algorithm-mmegolm_backupv1curve25519-aes-sha2
+//! [vodozemac documentation]: https://docs.rs/vodozemac/latest/vodozemac/pk_encryption/
+
+#![allow(missing_debug_implementations)]
+
+use matrix_sdk_crypto::vodozemac::{base64_encode, pk_encryption};
+use wasm_bindgen::prelude::*;
+
+use super::{Curve25519PublicKey, Curve25519SecretKey};
+
+/// A class representing an encrypted message using `PkEncryption`.
+#[wasm_bindgen]
+#[derive(Debug)]
+pub struct PkMessage {
+    inner: pk_encryption::Message,
+}
+
+/// The base64-encoded variant of a `PkMessage`.
+///
+/// This can be useful if the encrypted message should be put into JSON.
+#[wasm_bindgen(getter_with_clone)]
+pub struct Base64EncodedPkMessage {
+    /// The base64-encoded ciphertext.
+    pub ciphertext: String,
+    /// The base64-encoded message authentication code (MAC).
+    pub mac: String,
+    /// The base64-encoded ephemeral public key.
+    #[wasm_bindgen(js_name = "ephemeralKey")]
+    pub ephemeral_key: String,
+}
+
+#[wasm_bindgen]
+impl Base64EncodedPkMessage {
+    /// Creates a new base64-encoded encrypted message from its parts.
+    #[wasm_bindgen(constructor)]
+    pub fn new(ciphertext: &str, mac: &str, ephemeral_key: &str) -> Self {
+        Self {
+            ciphertext: ciphertext.to_owned(),
+            mac: mac.to_owned(),
+            ephemeral_key: ephemeral_key.to_owned(),
+        }
+    }
+}
+
+#[wasm_bindgen]
+impl PkMessage {
+    /// Returns the raw ciphertext as a byte vector.
```

Comment:
s/byte vector/`Uint8Array`/

likewise below.

Reply:
Done in: 6d80ac7f3d25d97ef06ad10c6f3d585a1cd2b17b.